### PR TITLE
feat: integrate MapStruct and Lombok for DTO mapping

### PR DESCRIPTION
**Pull request by matheusgeres**
- Add MapStruct and Lombok dependencies in pom.xml for DTO mapping and code simplification.
- Implement AddressDTOMapper using MapStruct for Address entity and DTO conversion.
- Refactor AddressService to use AddressDTOMapper for DTO to entity conversion.
- Introduce nickname field in AddressDTO and corresponding apelido field in Address entity for mapping demonstration.
- Setup maven-compiler-plugin with annotationProcessorPaths for MapStruct and Lombok processing.
- Remove unused imports and annotations in AddressService.
- Code formatting adjustments in User, OrderDTO, and User entities.

### CODE REVIEW

File: BookPlace/BookPlace/pom.xml [hunk 1]
```diff
@@ -85,6 +79,14 @@
...
 		</dependency>
 
+		<!-- INICIO /  Mapstruct -->
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>1.5.5.Final</version>
+		</dependency>
+		<!-- FIM /  Mapstruct -->
+
```
Comment (FrankDestro) [inline BookPlace/BookPlace/pom.xml:89]: o que o mapstruct faz ?

Comment (matheusgeres) [reply BookPlace/BookPlace/pom.xml:89]: Ele cria mappers automaticamente.

File: BookPlace/BookPlace/src/main/java/com/dev/BookPlace/dto/AddressDTO.java [hunk 2]
```diff
@@ -23,6 +23,7 @@ public class AddressDTO {
     private String country;
     private String postalCode;
     private Boolean main;
+    private String nickname;
```
Comment (FrankDestro) [inline BookPlace/BookPlace/src/main/java/com/dev/BookPlace/dto/AddressDTO.java:26]: o que é nickiname ?

File: BookPlace/BookPlace/src/main/java/com/dev/BookPlace/entities/bookplace/entities/Address.java [hunk 3]
```diff
@@ -27,6 +27,7 @@ public class Address{
     private String country;
     private String postalCode;
     private Boolean main;
+    private String apelido;
```
Comment (FrankDestro) [inline BookPlace/BookPlace/src/main/java/com/dev/BookPlace/entities/bookplace/entities/Address.java:30]: o que é apelido.

File: BookPlace/BookPlace/src/main/java/com/dev/BookPlace/mappers/AddressDTOMapper.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.dev.BookPlace.mappers;
+
+import com.dev.BookPlace.dto.AddressDTO;
+import com.dev.BookPlace.entities.bookplace.entities.Address;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(componentModel="spring")
+public interface AddressDTOMapper {
```
Comment (FrankDestro) [inline BookPlace/BookPlace/src/main/java/com/dev/BookPlace/mappers/AddressDTOMapper.java:9]: porque publico ?

**Review by FrankDestro — APPROVED** [review]
Obrigado voce é o cara

**Pull request merged**